### PR TITLE
refactor(sign in): removed shopify buy deps from signin api

### PR DESCRIPTION
**Pull request by aureate-labs-team**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/components/SearchResults.vue [hunk 1]
```diff
@@ -43,24 +43,62 @@
...
+                        :src="imageSlotProps.image"
+                        :width="imageSlotProps.imageWidth"
+                        class="sf-product-card__image lol"
+                      />
+                    </SfButton>
+                  </template>
                   <template slot="title">
                     <SfButton
-                      :link="localePath(getProductLink(product))"
+                      :link="localePath(
```
Comment (jeffpdotone) [inline packages/theme/components/SearchResults.vue:91]: Same here please use the route object I think it's already inside that function.

File: packages/theme/components/SearchResults.vue [hunk 2]
```diff
@@ -80,24 +118,64 @@
...
                 :score-rating="productGetters.getAverageRating(product)"
                 :image="productGetters.getCoverImage(product)"
+                :image-width="$device.isDesktopOrTablet ? 212 : 154"
+                :image-height="$device.isDesktopOrTablet ? 320 : 232"
                 :alt="productGetters.getName(product)"
                 :title="productGetters.getName(product)"
                 :add-to-cart-disabled="getStockCount(product) <= 0"
-                :link="localePath(getProductLink(product))"
+                :link="localePath(
+                  `/p/${productGetters.getId(product)}/${productGetters.getSlug(
```
Comment (jeffpdotone) [inline packages/theme/components/SearchResults.vue:127]: route object

File: packages/theme/components/SearchResults.vue [hunk 3]
```diff
@@ -80,24 +118,64 @@
...
+                      :width="imageSlotProps.imageWidth"
+                      class="sf-product-card__image lol"
+                    />
+                  </SfButton>
+                </template>
                 <template slot="title">
                   <SfButton
-                    :link="localePath(getProductLink(product))"
+                    :link="localePath(
+                      `/p/${productGetters.getId(product)}/${productGetters.getSlug(
```
Comment (jeffpdotone) [inline packages/theme/components/SearchResults.vue:167]: route object

File: packages/theme/components/SearchResults.vue [hunk 4]
```diff
@@ -297,33 +424,26 @@ export default {
       });
     };
 
-    const getProductLink = (product) => {
-      if (!product?.id || product?._slug) return '';
+    // const getProductLink = (product) => {
```
Comment (jeffpdotone) [inline packages/theme/components/SearchResults.vue:427]: I am not sure why this getting commented?

**Review by jeffpdotone — CHANGES_REQUESTED** [review]
can we rebased this I think there's too many unrelated changes to the topic.

File: docs/guide/checkout.md [hunk 5]
```diff
@@ -31,14 +31,15 @@ handle (url key) or product id.
...
 import { onSSR } from '@vue-storefront/core';
 import { useProduct, productGetters } from '@vue-storefront/shopify';
 
 export default {
-  setup(props, context) {
+  setup() {
+    const route = useRoute();
     const { products, search } = useProduct('products');
-    const { slug } = context.root.$route.params;
+    const { slug } = route?.value?.params;
```
[the remark this answers is not in the document]
Comment (aureate-labs-team) [reply docs/guide/checkout.md:42]: Yeah, it's giving the value as an object.

File: packages/theme/components/AppFooter.vue [hunk 6]
```diff
@@ -1,40 +1,64 @@
...
+          :key="item"
+        >
+          <SfMenuItem
+            :label="$t(item)"
+          />
         </SfListItem>
       </SfList>
     </SfFooterColumn>
-    <SfFooterColumn :title="$t('Social')">
+    <SfFooterColumn title="Social">
```
[the remark this answers is not in the document]
Comment (aureate-labs-team) [reply packages/theme/components/AppFooter.vue:55]: Resolved in #308

File: packages/theme/components/AppFooter.vue [hunk 7]
```diff
@@ -76,13 +108,13 @@ export default {
...
     };
-  }
+  },
 };
 </script>
 
 <style lang="scss">
 .footer {
-  margin-bottom: var(--spacer-xl);
+  margin-bottom: 3.75rem;
```
[the remark this answers is not in the document]
Comment (aureate-labs-team) [reply packages/theme/components/AppFooter.vue:117]: Resolved in #308

File: packages/theme/components/SearchResults.vue [hunk 8]
```diff
@@ -43,24 +43,62 @@
...
                   :score-rating="productGetters.getAverageRating(product)"
                   :image="productGetters.getCoverImage(product)"
+                  :image-width="$device.isDesktopOrTablet ? 212 : 154"
+                  :image-height="$device.isDesktopOrTablet ? 320 : 232"
                   :alt="productGetters.getName(product)"
                   :title="productGetters.getName(product)"
                   :add-to-cart-disabled="getStockCount(product) <= 0"
-                  :link="localePath(getProductLink(product))"
+                  :link="localePath(
+                    `/p/${productGetters.getId(product)}/${productGetters.getSlug(
```
[the remark this answers is not in the document]
Comment (aureate-labs-team) [reply packages/theme/components/SearchResults.vue:52]: Resolved in #308

File: packages/theme/components/SearchResults.vue (hunk 1 above)
Comment (aureate-labs-team) [reply packages/theme/components/SearchResults.vue:91]: Resolved in #308

File: packages/theme/components/SearchResults.vue (hunk 2 above)
Comment (aureate-labs-team) [reply packages/theme/components/SearchResults.vue:127]: Resolved in #308

File: packages/theme/components/SearchResults.vue (hunk 3 above)
Comment (aureate-labs-team) [reply packages/theme/components/SearchResults.vue:167]: Resolved in #308

File: packages/theme/components/SearchResults.vue (hunk 4 above)
Comment (aureate-labs-team) [reply packages/theme/components/SearchResults.vue:427]: Reverted. There was a condition bug so that the URL was not generating. It was fixed in #308 .

**Comment by aureate-labs-team** [comment]
> can we rebased this I think there's too many unrelated changes to the topic.

Its because we've an open PR #308

**Review by jeffpdotone — APPROVED** [review]

**Pull request merged**